h/static/scripts/controllers.coffee

```diff
@@ -48,7 +48,8 @@ class AppController
           annotationMapper.loadAnnotations data
         when 'delete'
           for annotation in data
-            $scope.$emit('annotationDeleted', annotation)
+            if a = threading.idTable[annotation.id]?.message
+              $scope.$emit('annotationDeleted', a)
```

Comment:
This would be the place to add it: https://github.com/hypothesis/h/blob/master/tests/js/controllers-test.coffee#L60-L71

You'll need to mockup the threading service and call the `streamer.onmessage()` handler with appropriate data and assert the behaviours you're looking for, in this case that emit is called on scope, and the annotation is fetched from the threading service.